### PR TITLE
William/connector cleanup

### PR DESCRIPTION
**Pull request by swansontec**
Built on top of #2729.

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

### CODE REVIEW

**Review by thehobbit85 — CHANGES_REQUESTED** [review]

File: src/components/scenes/CreateWalletAccountSelectScene.js [hunk 1]
```diff
@@ -435,8 +441,8 @@ export const CreateWalletAccountSelectScene = connect(
     fetchWalletAccountActivationPaymentInfo(paymentInfo: AccountPaymentParams, createdCoreWallet: EdgeCurrencyWallet) {
       dispatch(fetchWalletAccountActivationPaymentInfo(paymentInfo, createdCoreWallet))
     },
-    createAccountBasedWallet(walletName: string, walletType: string, fiatCurrencyCode: string, popScene: boolean, selectWallet: boolean) {
-      dispatch(createCurrencyWallet(walletName, walletType, fiatCurrencyCode, popScene, selectWallet))
+    async createAccountBasedWallet(walletName: string, walletType: string, fiatCurrencyCode: string, popScene: boolean, selectWallet: boolean) {
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/scenes/CreateWalletAccountSelectScene.js:443]: Nobody even awaits this, so it's no good.

File: src/components/scenes/FioAddressRegisterScene.js [hunk 2]
```diff
@@ -566,17 +566,15 @@ const getStyles = cacheStyles((theme: Theme) => ({
...
+export const FioAddressRegisterScene = connect(
+  (state: RootState): StateProps => ({
+    fioWallets: state.ui.wallets.fioWallets,
+    fioPlugin: state.core.account.currencyConfig[CURRENCY_PLUGIN_NAMES.FIO] ?? typeHack,
+    isConnected: state.network.isConnected
+  }),
   (dispatch: Dispatch): DispatchProps => ({
-    createFioWallet: () => dispatch(createFioWallet())
+    createFioWallet() {
+      return dispatch(createFioWallet())
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/scenes/FioAddressRegisterScene.js:575]: Could add async/await for clarity.

File: src/components/scenes/FioDomainRegisterScene.js [hunk 3]
```diff
@@ -330,11 +331,15 @@ const getStyles = cacheStyles((theme: Theme) => ({
...
+  (state: RootState): StateProps => ({
     fioWallets: state.ui.wallets.fioWallets,
     fioPlugin: state.core.account.currencyConfig[CURRENCY_PLUGIN_NAMES.FIO] ?? typeHack,
     isConnected: state.network.isConnected
-  }
-}, {})(withTheme(FioDomainRegister))
-export { FioDomainRegisterScene }
+  }),
+  (dispatch: Dispatch): DispatchProps => ({
+    createFioWallet() {
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/scenes/FioDomainRegisterScene.js:340]: Could add async/await for clarity.

File: src/components/scenes/RequestScene.js [hunk 4]
```diff
@@ -639,11 +639,15 @@ export const Request = connect(
...
-  (dispatch: Dispatch): RequestDispatchProps => ({
-    refreshReceiveAddressRequest: (walletId: string) => {
+  (dispatch: Dispatch): DispatchProps => ({
+    refreshReceiveAddressRequest(walletId: string) {
       dispatch(refreshReceiveAddressRequest(walletId))
     },
-    refreshAllFioAddresses: () => dispatch(refreshAllFioAddresses()),
-    onSelectWallet: (walletId: string, currencyCode: string) => dispatch(selectWalletFromModal(walletId, currencyCode))
+    refreshAllFioAddresses() {
+      return dispatch(refreshAllFioAddresses())
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/scenes/RequestScene.js:646]: Just await the result, no return.

File: src/components/scenes/TransactionsExportScene.js [hunk 5]
```diff
@@ -276,9 +276,9 @@ class TransactionsExportSceneComponent extends React.PureComponent<Props, State>
...
-  const denominationObject = getDisplayDenomination(state, ownProps.currencyCode)
-  return {
-    multiplier: denominationObject.multiplier
-  }
-})(withTheme(TransactionsExportSceneComponent))
+export const TransactionsExportScene = connect(
+  (state: RootState, ownProps: OwnProps): StateProps => ({
+    multiplier: getDisplayDenomination(state, ownProps.currencyCode).multiplier
+  }),
+  null
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/scenes/TransactionsExportScene.js]: Change all these to `(dispatch: DispatchProps) => ({})`

File: src/components/themed/SelectFioAddress.js [hunk 6]
```diff
@@ -262,35 +262,34 @@ class SelectFioAddressComponent extends React.PureComponent<Props, LocalState> {
...
+
+    if (!guiWallet || !currencyCode) {
+      return {
+        loading: true,
+        fioAddresses,
+        fioWallets,
+        currencyCode,
+        selectedWallet: guiWallet
+      }
+    }
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/themed/SelectFioAddress.js:279]: Derp, could combine these into one.

File: src/components/themed/WalletList.js [hunk 7]
```diff
@@ -250,7 +250,7 @@ class WalletListComponent extends React.PureComponent<Props> {
   }
 
   renderRow = (data: FlatListItem<WalletListItem>, rowMap: { [string]: SwipeRow }) => {
-    const { isModal, onPress, showSlidingTutorial, wallets } = this.props
+    const { isModal, onPress, selectWallet, showSlidingTutorial, wallets } = this.props
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/themed/WalletList.js:252]: Something about creating tokens with a different wallet selected than the one you want, it should crash / fail in some way.

File: src/components/scenes/RequestScene.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (swansontec) [reply src/components/scenes/RequestScene.js:646]: Nope, it's never awaited, so we can make it return `void` instead.

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**